### ISSUE TITLE
No line break after links

### ISSUE THREAD

**Issue by narfg** [servo/servo#5203]
The following test case
```html
<html>
<head>
<title>Line break</title>
</head>
<body>
Test1:
<div style="width: 70px; background-color: #00FF00;">
<a href="">link</a> lorem ipsum<br>
</div>
Test2:
<div style="width: 70px; background-color: #00FF00;">
link lorem ipsum
</div>
</body>
</html>
```
produces this output:
![outmin](https://cloud.githubusercontent.com/assets/158917/6616350/672dee34-c8ad-11e4-9c2d-24fcfa937924.png)
Instead of a line break after the link, the text is squished to the link and overflows the div. This only happens if the div surrounding the text is just a bit too small for both the link and the first word of the following text.
I noticed that this also breaks the output of tests/html/acid1.html
![outacid](https://cloud.githubusercontent.com/assets/158917/6616349/672d0cd0-c8ad-11e4-8e7a-5c5ff465a5b6.png)

**Comment by cojennin** [comment]
I'm unable to recreate this issue when building with [latest commit](https://github.com/servo/servo/commit/87af57cefc671f445c5b977f2b27d7ca9e70e390). 

I was able to recreate this issue a few days ago with an older build. In order to recreate the issue I had to change the font-family from using Verdana in tests/html/acid1.html.

**Comment by narfg** [comment]
Hmm, I'm still able to recreate the issue with the latest commit. Is there a way to create a test case that is independent of the font?

**Comment by narfg** [comment]
Fixed in #5493.
